octopusdeploy_framework/resource_project_group.go

```diff
@@ -0,0 +1,133 @@
+package octopusdeploy_framework
+
+import (
+	"context"
+	"fmt"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/projectgroups"
+	"github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy_framework/schemas"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+type projectGroupTypeResource struct {
+	*Config
+}
+
+func NewProjectGroupResource() resource.Resource {
+	return &projectGroupTypeResource{}
+}
+
+func (r *projectGroupTypeResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = ProviderTypeName + "_project_group"
+}
+
+func (r *projectGroupTypeResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: schemas.GetProjectGroupResourceSchema(),
+	}
+}
+
+func (r *projectGroupTypeResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	r.Config = ResourceConfiguration(req, resp)
+}
+
+func (r *projectGroupTypeResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *schemas.ProjectGroupTypeResourceModel
```

Comment:
My lack of go understanding will shine through here. Is there a reason this is a pointer while the other read, delete update methods are not?

Reply:
yeah, I think you are right, I will fix that